Formula/ices0.rb

```diff
@@ -0,0 +1,47 @@
+class Ices0 < Formula
+  desc "Source client for broadcasting to Icecast/Shoutcast servers in MP3 format"
+  homepage "https://github.com/Moonbase59/ices0"
+  url "https://github.com/Moonbase59/ices0/archive/v0.4.11.tar.gz"
+  sha256 "9179496149e763a75fea37bf6ec12947cad4bdb868b401d34728b353b836047a"
+  head "https://github.com/Moonbase59/ices0.git"
+
+  # building
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+  depends_on "pkg-config" => :build
+
+  # input/output
+  depends_on "faad2"
+  depends_on "flac"
+  depends_on "lame"
+  depends_on "libogg"
+  depends_on "libshout"
+  depends_on "libvorbis"
+  depends_on "libxml2"
+  depends_on "mp4v2"
+
+  def install
+    system "aclocal"
+    system "autoreconf", "-fi"
+    system "automake", "--add-missing"
+    system "./configure", "--prefix=#{prefix}"
+    system "make"
+    system "make", "install"
+  end
+
+  test do
+    # `test do` will create, run in and delete a temporary directory.
+    #
+    # For Homebrew/homebrew-core this will need to be a test that verifies
+    # the functionality of the software. Run the test with `brew test loudgain`.
+    # Options passed to `brew install` such as `--HEAD` also need to be
+    # provided to `brew test`.
+    #
+    # The installed folder is not in the path, so use the entire path to any
+    # executables being tested: `system "#{bin}/program", "do", "something"`.
+    # assert_match /^loudgain\s+#{version}.*/,
+    assert_match /^ices\s+0\.4\.11.*/,
+      shell_output("#{bin}/ices -V").strip
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I found many many formulae in core that just rely on a version check, so I assumed that should be enough to get things started. Problem with a loudness scanner/tagger is that one would have to upload large audio files to arrive at values that could be asserted.